buildbot.mariadb.org/master.cfg

```diff
@@ -530,9 +530,9 @@ f_tarball.addStep(steps.SetPropertyFromCommand(command="ls -1 *.tar.gz", extract
 #    masterdest=util.Interpolate('/srv/buildbot/packages/' + '%(prop:buildnumber)s'), url=util.Interpolate('https://ci.mariadb.org/' + "%(prop:buildnumber)s"), urlText="Download", doStepIf=hasFiles))
 f_tarball.addStep(steps.ShellCommand(name='save_packages', haltOnFailure=True, command=util.Interpolate('cp -r ' + '%(prop:builddir)s' + '/build/mkdist/' + '%(prop:buildnumber)s' + ' /packages && sync /packages/' + '%(prop:buildnumber)s')))
 f_tarball.addStep(steps.Trigger(schedulerNames=['s_protected_branches'], waitForFinish=False, updateSourceStamp=False, doStepIf=waitIfStaging,
-    set_properties={"tarbuildnum" : Property("buildnumber"), "mariadb_version" : Property("mariadb_version"), "master_branch" : Property("master_branch")}))
+    set_properties={"tarbuildnum" : Property("tarbuildnum"), "mariadb_version" : Property("mariadb_version"), "master_branch" : Property("master_branch")}))
 f_tarball.addStep(steps.Trigger(schedulerNames=['s_upstream_all'], waitForFinish=False, updateSourceStamp=False,
-    set_properties={"tarbuildnum" : Property("buildnumber"), "mariadb_version" : Property("mariadb_version"), "master_branch" : Property("master_branch")}))
+    set_properties={"tarbuildnum" : Property("tarbuildnum"), "mariadb_version" : Property("mariadb_version"), "master_branch" : Property("master_branch")}))
```

Comment:
The tarbuildnum is correctly set without any changes. Based on this PR it will be undefined since the tarball builder doesn't have it set. So, we need to set tarbuildnum as buildnumber. This happens in the tarball builder and sets the property tarbuildnum for all triggered builders with the current buildnumer of the tarball builder.

While indeed in the log, there seems to be an issue, looking at the same build that you point, you can see that the tarbuildnum is correctly set (see below)

![Screenshot 2022-01-06 at 16 09 26](https://user-images.githubusercontent.com/5010835/148395651-8dbdb267-573e-4a15-8582-d392c628464d.png)

So, I susptect to be an issue where the deb scripts looks at the buildnumber and not at the tarbuildnum